crates/typed-store/src/rocks/mod.rs

```diff
@@ -94,11 +112,40 @@ pub struct DBMap<K, V> {
     db_metrics: Arc<DBMetrics>,
     read_sample_interval: SamplingInterval,
     write_sample_interval: SamplingInterval,
+    _cf_metric_reporter: Arc<JoinHandle<()>>,
+    exit_signal: Arc<AtomicBool>,
 }
 
 unsafe impl<K: Send, V: Send> Send for DBMap<K, V> {}
 
 impl<K, V> DBMap<K, V> {
+    pub(crate) fn new(
+        db: Arc<rocksdb::DBWithThreadMode<MultiThreaded>>,
+        opt_cf: &str,
+        db_metrics: Arc<DBMetrics>,
+    ) -> Self {
+        let exit = Arc::new(AtomicBool::new(false));
+        let exit_cloned = exit.clone();
+        let db_cloned = db.clone();
+        let db_metrics_cloned = db_metrics.clone();
+        let cf = opt_cf.to_string();
+        let _cf_metric_reporter = Arc::new(std::thread::spawn(move || {
+            while !exit_cloned.load(Ordering::Relaxed) {
+                Self::report_metrics(&db, &cf, &db_metrics);
+                sleep(Duration::from_millis(CF_METRICS_REPORT_PERIOD_MILLIS));
```

Comment:
Assuming that `Self::report_metrics` is not too expensive in terms of CPU (it seems it just reads some variables from rocksdb), it might be better to have this reporting thread to be a tokio task on the main scheduler.

So instead of using `thread::spawn` it seem to be more reasonable to use `tokio::spawn` and then use `tokio::sleep` here. This way we will avoid creating new thread for each DBMap

Reply:
This I am not 100% certain of. It is likely better to use spawn_blocking to be safe.

Reply:
I see. if you do want a blocking thread then `thread::spawn` probably makes more sense (spawn_blocking might have some additional overhead on thread housekeeping that we don't care about)